src/jvmTest/kotlin/bdd/SpaceEngineersCucumberTest.kt

```diff
@@ -100,6 +130,55 @@ class SpaceEngineersCucumberTest {
         )
     }
 
+    private fun observeBlocks(): List<SeBlock> {
+        return environment.observe(ObservationArgs(ObservationMode.BLOCKS)).allBlocks
+    }
+
+    private fun blockToGrind(): SeBlock {
+        return observeBlocks().first { it.id == context.lastNewBlockId }
+    }
+
+    @When("Character grinds to {double} integrity.")
+    fun character_grinds_to_integrity(integrity: Double) {
+        var currentIntegrity = blockToGrind().integrity
+        environment.equip(context.grinderLocation!!)
+        sleep(500)
+        environment.interact(InteractionArgs(InteractionType.BEGIN_USE))
+        while (currentIntegrity > integrity) {
+            currentIntegrity = blockToGrind().integrity
+        }
+        environment.interact(InteractionArgs(InteractionType.END_USE))
+    }
+
+    @When("Character grinds to {double}% integrity.")
+    fun character_grinds_until_to_integrity_percentage(percentage: Double) {
+        val blockToGrind = blockToGrind()
+        var currentIntegrity = blockToGrind.integrity
+        val integrity = blockToGrind.maxIntegrity * percentage * 0.01
+        environment.equip(context.grinderLocation!!)
+        sleep(500)
+        environment.interact(InteractionArgs(InteractionType.BEGIN_USE))
+        while (currentIntegrity > integrity) {
+            currentIntegrity = blockToGrind().integrity
+        }
+        environment.interact(InteractionArgs(InteractionType.END_USE))
```

Comment:
It would be nice to limit the loop somehow. (Max number of repetitions?)

Or even smarter would be to abort if the integrity is not moving at all for some time.

I understand it would be not a trivial fix.